### DISCUSSION TITLE
Superset Visualizations are timing out after 60 seconds. Want to increase that. How can I implement it?

### DISCUSSION THREAD

**Discussion by imamaigein99** [apache/superset#19561] [Q&A / Help]
[](https://superset.apache.org/docs/miscellaneous/issue-codes#issue-1000)[](https://superset.apache.org/docs/miscellaneous/issue-codes#issue-1001)Timeout error
We’re having trouble loading this visualization. Queries are set to timeout after 60 seconds.

This may be triggered by:
Issue 1000 - The dataset is too large to query. 
Issue 1001 - The database is under an unusual load.

**Comment by srinify** [comment]
Hey there @imamaigein99 moving to Github Discussions as this isn't a bug per say!

In general, I would think about Superset as a light frontend over your existing database / data stack. So if you attempt to query a large # of rows / records that your database can't handle, it'll take too long for the results to come back.

**Reply by igorlukanin** [reply]
I'd like to double down on what @srinify shared. If the upstream part of the data stack (e.g., a database) is not able to handle the load, it would probably work better to optimize that part rather than to increase the timeouts.

I would also recommend to check out this [blog post](https://preset.io/blog/metrics-dashboard-with-superset-cube/), especially the section called "Making business intelligence fast ⚡". In that blog post, I explained how Cube can help increase the performance of practically any data source and allow dashboards in Superset to load in under a second. Hope it helps!
